constellations/gemini-admin/src/user/userAFH.js

```diff
@@ -0,0 +1,180 @@
+"use strict";
+require("./userAFH.sass");
+import React from "react";
+import moment from "moment";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import API from "../api.js";
+
+export class UserAFHPage extends React.Component {
+    state = {
+        id: 0,
+        user: {},
+        userAFHs: [],
+        afhs: [],
+        afhId: 0,
+    };
+
+    componentDidMount = () => {
+        this.fetchUser();
+    };
+
+    fetchUser = () => {
+        const id = this.props.id;
+
+        const apiCalls = [
+            API.get("api/users/user/" + id),
+            API.get("api/userafhs/users/" + id),
+        ];
+
+        axios
+            .all(apiCalls)
+            .then(
+                axios.spread((...responses) => {
+                    let afhIds = [];
+                    responses[1].data.forEach((userAFH) => {
+                        afhIds.push(userAFH.afhId);
+                    });
+
+                    this.setState({
+                        id: id,
+                        user: responses[0].data,
+                    });
+
+                    if (afhIds.length > 0) {
+                        this.fetchAFHs(afhIds);
+                    }
+                })
+            )
+            .catch((err) => {
+                alert("Could not fetch user: " + err.response.data);
+            });
+    };
+
+    fetchAFHs = (afhIds) => {
+        let searchArray = new Set(afhIds);
+        API.get("api/askforhelp/all")
+            .then((res) => {
+                var userAFHs = [];
+                var afhs = [];
+                res.data.forEach((afh) => {
+                    if (searchArray.has(afh.id)) {
+                        userAFHs.push(afh);
```

Comment:
Is afh here an afh object? Just wondering how that gets appended to an array that's for userAFHs.

Reply:
Yeah its the afh object that contains the userId and the afhId

Reply:
`userAFHs` is the afhs that the user is signed up for and `afhs` is all the others